mloop/utilities.py

```diff
@@ -464,6 +465,25 @@ def stop(self):
         '''
         pass
 
+class ParameterScaler(skp.MinMaxScaler):
+    def __init__(self, min_boundary, max_boundary, *args, **kwargs):
+        '''
+        Class to scale the parameters of the gaussian process. 
+        All parameters are mapped (by default) between [0, 1] using min and max boundaries
+        '''
+        if len(min_boundary) != len(max_boundary):
+            raise("The minimum and maximum boundary arrays must have the same lengths")
+
+        self.min_boundary = min_boundary
+        self.max_boundary = max_boundary
+        return super().__init__(*args, **kwargs)
+
+    def partial_fit(self, X=None, *args, **kwargs):
+        '''
+        Teach the scaler that we want to scale things based on the minimum and maximum boundaries
+        '''
+        X = [self.min_boundary, self.max_boundary]  # Maybe need transpose of this.
```

Comment:
We can ditch this comment now if you've checked that this is the form expected by `super().partial_fit()`